packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -384,6 +384,9 @@ export function useAsyncData<
         await asyncData.refresh()
       }
     })
+    if (instance) {
+      onUnmounted(off)
+    }
     if (hasScope) {
```

Comment:
```suggestion
    } else if (hasScope) {
```

Reply:
we should also apply the same fix to L378-380

Reply:
What if it has both instance and scope?

Reply:
we want to remove when it is unmounted - it will be reinitialised when component is mounted again. the scope is at a higher level (at nuxt app level, or within a pinia store).

Reply:
Thanks! Agreed then, will do a bit later

Reply:
Done!